[Bug]: fluent-ui v9 peer dependency warnings

### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
* WSL2
  System:
    OS: Linux 5.10 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (2) x64 Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz
    Memory: 15.56 GB / 19.54 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
```


### Are you reporting Accessibility issue?

no

### Reproduction

https://github.com/lnarimatsu/vite_fluentv9_issue.git

### Bug Description



## Issue
During a brand new minimal installation of Fluent Ui v9 on a brand new vite project I got the peer dependency overrides decribed below.

Commands I've run before these warning messages:
`npm create vite@latest brand_new_project -- --template react-ts`
`cd brand_new_project`
`npm install`
`npm install @fluentui/react-components`


## Actual Behavior
Warning messages related to peer dependencies conflicts and overrides.

```
npm WARN ERESOLVE overriding peer dependency
npm WARN ERESOLVE overriding peer dependency
npm WARN ERESOLVE overriding peer dependency
npm WARN ERESOLVE overriding peer dependency
```

## Expected Behavior
No NPM Warnings.... 

package.json
```json
{
  "name": "brand_new_project",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@fluentui/react-components": "^9.15.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@types/react": "^18.0.27",
    "@types/react-dom": "^18.0.10",
    "@vitejs/plugin-react": "^3.1.0",
    "typescript": "^4.9.3",
    "vite": "^4.1.0"
  }
}
```

### Logs

```shell
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN Found: @types/react@18.0.28
npm WARN node_modules/@types/react
npm WARN   dev @types/react@"^18.0.28" from the root project
npm WARN   44 more (@fluentui/react-accordion, @fluentui/react-alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @types/react@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN   @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN   node_modules/@fluentui/react-components
npm WARN
npm WARN Conflicting peer dependency: @types/react@17.0.53
npm WARN node_modules/@types/react
npm WARN   peer @types/react@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN   node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN     @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN     node_modules/@fluentui/react-components
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN Found: @types/react-dom@18.0.11
npm WARN node_modules/@types/react-dom
npm WARN   dev @types/react-dom@"^18.0.11" from the root project
npm WARN   41 more (@fluentui/react-accordion, @fluentui/react-alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @types/react-dom@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN   @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN   node_modules/@fluentui/react-components
npm WARN
npm WARN Conflicting peer dependency: @types/react-dom@17.0.19
npm WARN node_modules/@types/react-dom
npm WARN   peer @types/react-dom@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN   node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN     @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN     node_modules/@fluentui/react-components
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"^18.2.0" from the root project
npm WARN   48 more (@fluentui/react-accordion, @fluentui/react-alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN   @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN   node_modules/@fluentui/react-components
npm WARN
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN   node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN     @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN     node_modules/@fluentui/react-components
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   react-dom@"^18.2.0" from the root project
npm WARN   42 more (@fluentui/react-accordion, @fluentui/react-alert, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react-dom@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN   @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN   node_modules/@fluentui/react-components
npm WARN
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@">=16.8.0 <18.0.0" from @fluentui/react-virtualizer@9.0.0-alpha.6
npm WARN   node_modules/@fluentui/react-components/node_modules/@fluentui/react-virtualizer
npm WARN     @fluentui/react-virtualizer@"9.0.0-alpha.6" from @fluentui/react-components@9.15.3
npm WARN     node_modules/@fluentui/react-components
```


### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

--- a/change/@fluentui-react-virtualizer-f4f74084-b93e-4113-8e45-29229d62546e.json
+++ b/change/@fluentui-react-virtualizer-f4f74084-b93e-4113-8e45-29229d62546e.json
@@ -1,6 +1,6 @@
 {
   "type": "prerelease",
-  "comment": "fix peer deps of react virtualizer",
+  "comment": "fix: Updating peer dependencies to support React 18.",
   "packageName": "@fluentui/react-virtualizer",
   "email": "mgodbolt@microsoft.com",
   "dependentChangeType": "patch"

--- a/change/@fluentui-react-virtualizer-f4f74084-b93e-4113-8e45-29229d62546e.json
+++ b/change/@fluentui-react-virtualizer-f4f74084-b93e-4113-8e45-29229d62546e.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix peer deps of react virtualizer",
+  "packageName": "@fluentui/react-virtualizer",
+  "email": "mgodbolt@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/react-components/react-virtualizer/package.json
+++ b/packages/react-components/react-virtualizer/package.json
@@ -36,10 +36,10 @@
     "tslib": "^2.1.0"
   },
   "peerDependencies": {
-    "@types/react": ">=16.8.0 <18.0.0",
-    "@types/react-dom": ">=16.8.0 <18.0.0",
-    "react": ">=16.8.0 <18.0.0",
-    "react-dom": ">=16.8.0 <18.0.0"
+    "@types/react": ">=16.8.0 <19.0.0",
+    "@types/react-dom": ">=16.8.0 <19.0.0",
+    "react": ">=16.8.0 <19.0.0",
+    "react-dom": ">=16.8.0 <19.0.0"
   },
   "beachball": {
     "disallowedChangeTypes": [

